project/src/main/java/com/google/autograder/servlets/helpers/API.java

```diff
@@ -68,6 +71,27 @@ public static String getCurrentUserAPIAuthorization() {
         return authorization;
     }
 
+    public static HttpURLConnection getAuthenticatedRequest(String method, String endpoint) {
+        try {
+            String authorization = getCurrentUserAPIAuthorization();
+
+            if (authorization == null) {
+                return null;
+            }
+
+            HttpURLConnection connection = (HttpURLConnection) new URL(endpoint).openConnection();
+
+            connection.setRequestMethod(method);
+            connection.setRequestProperty("Accept", "application/json");
+            connection.setRequestProperty("Authorization", authorization);
+
+            return connection;
+        } catch(Exception e) {
+            e.printStackTrace();
```

Comment:
This would be a good time to learn how to properly set up logging (https://cloud.google.com/appengine/docs/standard/java/logs#writing_application_logs). The reason we typically don't use `System.out.println` or `e.printStackTrace` in production applications is that it's not flexible and usually doesn't format the messages in an easy-to-read way. If you use standard logging libraries then they'll take care of formatting messages so that it includes the timestamp and other metadata as well as ensuring that they'll show up in the right places (in the console, saved in some file somewhere, and/or other methods of persistence)

For this you'd do something like:
```
class API {
  private static final Logger logger = Logger.getLogger(API.class.getName()); // This makes all messages logged from this class have the name "API" so it's easier to see where they're from

  public static HttpURLConnection getAuthenticatedRequest(...) {
    try {
      ...
    } catch (Exception e) {
      logger.log(Level.SEVERE, "Error getting authenticated request.", e);
    }
  }
}
```

Reply:
Interesting, thank you. +1